test/suite/jxui/jxui.go

```diff
@@ -0,0 +1,278 @@
+package jxui
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/v28/github"
+	"github.com/jenkins-x/bdd-jx/test/helpers"
+	"github.com/jenkins-x/bdd-jx/test/utils"
+	"github.com/jenkins-x/bdd-jx/test/utils/runner"
+	"github.com/jenkins-x/jx/pkg/gits"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type AppTestOptions struct {
+	helpers.TestOptions
+}
+
+var _ = Describe("Jenkins X UI tests", func() {
+	var appTestOptions AppTestOptions
+
+	BeforeEach(func() {
+		appTestOptions = AppTestOptions{
+			helpers.TestOptions{
+				ApplicationName: helpers.TempDirPrefix + "ui" + "-" + strconv.FormatInt(GinkgoRandomSeed(), 10),
+				WorkDir:         helpers.WorkDir,
+			},
+		}
+		if !appTestOptions.GitOpsEnabled() {
+			Skip("Skipping apps tests for UI since they require a gitops setup")
+		}
+	})
+
+	_ = appTestOptions.UITest()
+})
+
+func (t *AppTestOptions) UITest() bool {
+	var (
+		jxHome       string
+		ctx          context.Context
+		gitHubClient *github.Client
+		gitInfo      *gits.GitRepository
+		err          error
+	)
+
+	BeforeEach(func() {
+		By("setting a temporary JX_HOME directory")
+		jxHome, err = ioutil.TempDir("", helpers.TempDirPrefix+"ui-jx-home-")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		_ = os.Setenv("JX_HOME", jxHome)
+		utils.LogInfo(fmt.Sprintf("Using '%s' as JX_HOME", jxHome))
+	})
+
+	BeforeEach(func() {
+		By("setting the GitHub token")
+		helpers.SetGitHubToken()
+	})
+
+	BeforeEach(func() {
+		By("setting up a GitHub client")
+		ctx = context.Background()
+		gitHubClient = t.GitHubClient()
+	})
+
+	BeforeEach(func() {
+		By("parsing the gitops dev repo information")
+		gitInfo, err = gits.ParseGitURL(t.GitOpsDevRepo())
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		_ = os.RemoveAll(jxHome)
+	})
+
+	return Context("UI", func() {
+		var uiURL = ""
+		It("ensure UI is not installed", func() {
+			pr, err := helpers.GetPullRequestWithTitle(gitHubClient, ctx, gitInfo.Organisation, gitInfo.Name, fmt.Sprintf("Add %s %s", uiAppName, uiAppVersion))
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(pr).Should(BeNil())
+		})
+
+		It("install UI via 'jx add app'", func() {
```

Comment:
Now we install the ui app twice, once for jxui and once for the app life cycle, right? would it make sense to test the app life cycle with a different app?

Reply:
App lifecycle is to validate any app of if we have another candidate we can take it.
Just take into account that jxui test suite will not run upstream, but that's what we discussed and agreed.